src/backend/distributed/commands/schema.c

```diff
@@ -53,76 +85,48 @@ PreprocessDropSchemaStmt(Node *node, const char *queryString,
 						 ProcessUtilityContext processUtilityContext)
 {
 	DropStmt *dropStatement = castNode(DropStmt, node);
-	Relation pgClass = NULL;
-	HeapTuple heapTuple = NULL;
-	SysScanDesc scanDescriptor = NULL;
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 1;
-	Oid scanIndexId = InvalidOid;
-	bool useIndex = false;
+	Assert(dropStatement->removeType == OBJECT_SCHEMA);
 
-	if (dropStatement->behavior != DROP_CASCADE)
+	if (!ShouldPropagate())
 	{
 		return NIL;
 	}
 
-	Value *schemaValue = NULL;
-	foreach_ptr(schemaValue, dropStatement->objects)
+	EnsureCoordinator();
+
+	List *distributedSchemas = FilterDistributedSchemas(dropStatement->objects);
+
+	if (list_length(distributedSchemas) < 1)
 	{
-		const char *schemaString = strVal(schemaValue);
-		Oid namespaceOid = get_namespace_oid(schemaString, true);
+		return NIL;
+	}
 
-		if (namespaceOid == InvalidOid)
-		{
-			continue;
-		}
+	EnsureSequentialModeForSchemaDDL();
 
-		pgClass = table_open(RelationRelationId, AccessShareLock);
+	Value *schemaVal = NULL;
+	foreach_ptr(schemaVal, distributedSchemas)
+	{
+		InvalidateFKeysForDistributedTablesInSchema(strVal(schemaVal));
+	}
 
-		ScanKeyInit(&scanKey[0], Anum_pg_class_relnamespace, BTEqualStrategyNumber,
-					F_OIDEQ, namespaceOid);
-		scanDescriptor = systable_beginscan(pgClass, scanIndexId, useIndex, NULL,
-											scanKeyCount, scanKey);
+	/* save the original object list to use later */
+	List *originalObjects = dropStatement->objects;
 
-		heapTuple = systable_getnext(scanDescriptor);
-		while (HeapTupleIsValid(heapTuple))
-		{
-			Form_pg_class relationForm = (Form_pg_class) GETSTRUCT(heapTuple);
-			char *relationName = NameStr(relationForm->relname);
-			Oid relationId = get_relname_relid(relationName, namespaceOid);
-
-			/* we're not interested in non-valid, non-distributed relations */
-			if (relationId == InvalidOid || !IsCitusTable(relationId))
-			{
-				heapTuple = systable_getnext(scanDescriptor);
-				continue;
-			}
-
-			if (IsCitusTableType(relationId, REFERENCE_TABLE))
-			{
-				/* prevent concurrent EnsureReferenceTablesExistOnAllNodes */
-				int colocationId = CreateReferenceTableColocationId();
-				LockColocationId(colocationId, ExclusiveLock);
-			}
-
-			/* invalidate foreign key cache if the table involved in any foreign key */
-			if (TableReferenced(relationId) || TableReferencing(relationId))
-			{
-				MarkInvalidateForeignKeyGraph();
-
-				systable_endscan(scanDescriptor);
-				table_close(pgClass, NoLock);
-				return NIL;
-			}
+	/* modify object list such that it only contains the distribute ones */
+	dropStatement->objects = distributedSchemas;
 
-			heapTuple = systable_getnext(scanDescriptor);
-		}
+	/* deparse sql with the distributed ones */
```

Comment:
this comment is exactly telling what the code tells as well.

In general you don't have to document what it does, only why it does so if that is not obvious.
I would suggest removing this comment as it doesn't add value